privacy-on-beam/pbeam/sum.go

```diff
@@ -80,8 +80,19 @@ type SumParams struct {
 // SumPerKey transforms a PrivatePCollection<K,V> either into a
 // PCollection<K,int64> or a PCollection<K,float64>, depending on whether its
 // input is an integer type or a float type.
-func SumPerKey(s beam.Scope, pcol PrivatePCollection, params SumParams) beam.PCollection {
+func SumPerKey(s beam.Scope, pcol PrivatePCollection, params SumParams, partitions ...beam.PCollection) beam.PCollection {
+	if len(partitions) > 1 {
+		log.Exitf("Only one partition PCollection can be specified. %v were specified.", len(partitions))
+	}
+
 	s = s.Scope("pbeam.SumPerKey")
+	if len(partitions) == 1 {
+		if pcol.codec.KType.T != partitions[0].Type().Type() {
```

Comment:
Also here, are types guaranteed to be consistent within a pcol?

Reply:
From what I know, pcol.codec.KType.T should be the type of all of the partitions in the original pcol. Are you suggesting that I check that the type of every specified partition is equal to pcol.codec.KType.T?

Reply:
partitions[0] is the entire PCollection since partitions is input to the variadic function.

Reply:
ah makes sense. Thanks!